lib/posix-socket/events.h

```diff
@@ -126,12 +129,13 @@ static inline void uk_socket_evd_raddr_set_from(struct uk_socket_event_data
 				from_evd->raddr_len);
 }
 
-static inline void uk_socket_event_raise(struct uk_socket_event_data *evd,
-					 enum uk_socket_event_type event)
+static inline void socket_event_raise(struct uk_socket_event_data *evd,
+				      enum uk_socket_state new_state,
+				      struct uk_event *evt)
 {
 	struct uk_socket_event emsg = {
 		.id        = evd->id,
-		.event     = event,
+		.state     = evd->state,
```

Comment:
This new field and the new way how we use it, requires initialization in `uk_socket_evd_init()` **and** `uk_socket_evd_init_from()`
```diff
+       evd->state      = UK_SOCKET_STATE_CLOSE;
```

Can you add this line to both functions? Without it, I saw some garbage happening for freshly opened/created sockets when they emit an state change event the first time.

Reply:
Good catch 🙏🏻 ! Thanks, I'll fix it